### PR TITLE
Use proper escaping methods for attributes.

### PR DESCRIPTION
**Pull request by EreMaijala**
Includes some template cleanup especially for pagination.

### CODE REVIEW

File: module/VuFind/src/VuFind/Controller/MyResearchController.php [hunk 1]
```diff
@@ -2195,7 +2195,7 @@ protected function addPendingEmailChangeMessage($user)
                     'html' => true,
                     'msg' => 'email_change_pending_html',
                     'tokens' => [
-                        '%%pending%%' => $user->pending_email,
+                        '%%pending%%' => $this->escapeHtml($user->pending_email),
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Controller/MyResearchController.php]: Did you test this? I don't think escapeHtml() is defined in the context of controllers, though perhaps I am forgetting something.

File: themes/bootstrap3/templates/Recommend/AuthorFacets.phtml [hunk 2]
```diff
@@ -10,7 +10,7 @@
           </div>
           <div class="author-list">
         <?php endif; ?>
-        <a href="<?=$this->url('author-home') . '?author=' . urlencode($author['value'])?>"><?=$author['value'] ?><?php /* count disabled -- uncomment to add: echo ' - ' . $author['count']; */ ?></a>
+        <a href="<?=$this->url('author-home') . '?author=' . urlencode($author['value'])?>"><?=$this->escapeHtml($author['value']) ?><?php /* count disabled -- uncomment to add: echo ' - ' . $author['count']; */ ?></a>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/Recommend/AuthorFacets.phtml]: Should we escape the commented out example too while we're at it?

File: themes/bootstrap3/templates/RecordDriver/EDS/core.phtml [hunk 3]
```diff
@@ -109,11 +109,11 @@
           $icon = $customLink['Icon'] ?? '';
           $name = $customLink['Text'] ?? '';
         ?>
-        <span>
+        <div>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordDriver/EDS/core.phtml]: What's the motivation of this change? Seems unrelated to everything else.

File: themes/bootstrap3/templates/record/ajaxview-accordion.phtml [hunk 4]
```diff
@@ -74,12 +74,12 @@
         if ($this->defaultTab === strtolower($tab)) { $tab_classes[] = 'default'; }
       ?>
       <div class="panel panel-default <?=implode(' ', $tab_classes) ?>">
-        <div id="<?=strtolower($tab)?>_<?=$idSuffix?>" class="list-tab-toggle panel-heading" data-toggle="collapse" data-parent="#accordion_<?=$idSuffix?>" data-target="#<?=strtolower($tab)?>_<?=$idSuffix?>-content"<?php if ($obj->supportsAjax() && in_array($tab, $this->backgroundTabs)):?> data-background<?php endif ?>>
+        <div id="<?=$this->escapeHtmlAttr(strtolower($tab))?>_<?=$idSuffix?>" class="list-tab-toggle panel-heading" data-toggle="collapse" data-parent="#accordion_<?=$idSuffix?>" data-target="#<?=strtolower($tab)?>_<?=$idSuffix?>-content"<?php if ($obj->supportsAjax() && in_array($tab, $this->backgroundTabs)):?> data-background<?php endif ?>>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/record/ajaxview-accordion.phtml]: Don't we also need to escape $idSuffix? Would it be helpful to create a variable here due to the repetition below?

File: themes/bootstrap3/templates/record/ajaxview-tabs.phtml [hunk 5]
```diff
@@ -46,7 +46,7 @@
       if (!$obj->supportsAjax()) { $tab_classes[] = 'noajax'; }
     ?>
     <li<?=count($tab_classes) > 0 ? ' class="' . implode(' ', $tab_classes) . '"' : ''?>>
-      <a id="<?=strtolower($tab)?>_<?=$idSuffix?>" class="list-tab-toggle" href="<?=$this->recordLink()->getTabUrl($this->driver, $tab)?>#tabnav" data-toggle="tab" data-target="#<?=strtolower($tab)?>_<?=$idSuffix?>-content"<?php if ($obj->supportsAjax() && in_array($tab, $this->backgroundTabs)):?> data-background<?php endif ?>><?=$this->transEsc($desc)?></a>
+      <a id="<?=$this->escapeHtmlAttr(strtolower($tab))?>_<?=$idSuffix?>" class="list-tab-toggle" href="<?=$this->recordLink()->getTabUrl($this->driver, $tab)?>#tabnav" data-toggle="tab" data-target="#<?=strtolower($tab)?>_<?=$idSuffix?>-content"<?php if ($obj->supportsAjax() && in_array($tab, $this->backgroundTabs)):?> data-background<?php endif ?>><?=$this->transEsc($desc)?></a>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/record/ajaxview-tabs.phtml]: Another unescaped $idSuffix.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @EreMaijala. I had a few questions and comments about this one; once you've addressed them, I'll run the full test suite just to be sure nothing was missed/broken.

File: module/VuFind/src/VuFind/Controller/MyResearchController.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/Controller/MyResearchController.php]: Oops, no, I missed this one. There was similar code in a template and I mixed things up. I'll fix.

File: themes/bootstrap3/templates/record/ajaxview-accordion.phtml (hunk 4 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/record/ajaxview-accordion.phtml]: $idSuffix is already escaped. I'll change the name to indicate that.

Comment (EreMaijala) [reply themes/bootstrap3/templates/record/ajaxview-accordion.phtml]: The one below is not the same, it has an additional `-content` suffix, so I'd keep them separate.

File: themes/bootstrap3/templates/RecordDriver/EDS/core.phtml (hunk 3 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/RecordDriver/EDS/core.phtml]: It's indeed unrelated, now separated to #2065.

File: themes/bootstrap3/templates/record/ajaxview-tabs.phtml (hunk 5 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/record/ajaxview-tabs.phtml]: This is escaped as well, but the name didn't indicate it. Now changed.

File: themes/bootstrap3/templates/Recommend/AuthorFacets.phtml (hunk 2 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/Recommend/AuthorFacets.phtml]: Done.

File: module/VuFind/src/VuFind/Controller/MyResearchController.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/Controller/MyResearchController.php]: Fixed.

File: themes/bootstrap3/templates/record/ajaxview-accordion.phtml (hunk 4 above)
Comment (demiankatz) [reply themes/bootstrap3/templates/record/ajaxview-accordion.phtml]: Agreed, I just thought it might make sense to have an $idPrefix variable to make the difference between the two ids more obvious (since the lines are currently quite long). Fine to leave as-is, though!

**Comment by demiankatz** [comment]
Thanks for the fixes, @EreMaijala -- this all looks good to me now, and the test suite is passing, so I'm going to merge it.
